### PR TITLE
Add screen and multiply blending modes

### PR DESCRIPTION
**Pull request by romainguy**
This change also fixes a sorting issue: different sorting modes
were sorted in different buckets which is incorrect. We want
to sort only by distance.

### CODE REVIEW

File: libs/filamat/src/shaders/ShaderGenerator.cpp [hunk 1]
```diff
@@ -254,16 +254,22 @@ const std::string ShaderGenerator::createFragmentProgram(filament::backend::Shad
...
+            cg.generateDefine(fs, "BLEND_MODE_MASKED", true);
+            break;
         case BlendingMode::FADE:
             // Fade is a special case of transparent
             cg.generateDefine(fs, "BLEND_MODE_TRANSPARENT", true);
             cg.generateDefine(fs, "BLEND_MODE_FADE", true);
             break;
-        case BlendingMode::ADD:
-            cg.generateDefine(fs, "BLEND_MODE_ADD", true);
+        case BlendingMode::MULTIPLY:
```
Comment (pixelflinger) [inline libs/filamat/src/shaders/ShaderGenerator.cpp]: There are no equivalent of these constants on the java side?

Comment (romainguy) [reply libs/filamat/src/shaders/ShaderGenerator.cpp]: Yeah I realized last night I forgot :)

**Comment by romainguy** [comment]
So much easier to go through git logs than remembering to update release notes :p

**Comment by romainguy** [comment]
Addressed comments

**Pull request merged**